packages/flutter/lib/src/material/arc.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart'; // @required definition
+
+abstract class _MaterialArc<T> {
```

Comment:
How is this different than Tween?  I guess the begin and end are immutable.  We could make a base class for Tween called UnmodifiableTween and inherit from that.

Alternatively, we could override the setters here to throw exceptions.  That might be the simpler change.